backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/push/parsing/RuleSetChangedEventParser.java

```diff
@@ -45,11 +47,14 @@ public Optional<RuleSetChangedEvent> parse(String jsonData) {
     return Optional.of(new RuleSetChangedEvent(
       payload.projects,
       payload.activatedRules.stream().map(changedRule -> new RuleSetChangedEvent.ActiveRule(
-        changedRule.key,
-        changedRule.language,
-        IssueSeverity.valueOf(changedRule.severity),
-        changedRule.params.stream().filter(p -> p.value != null).collect(Collectors.toMap(p -> p.key, p -> p.value)),
-        changedRule.templateKey))
+          changedRule.key,
+          changedRule.language,
+          IssueSeverity.valueOf(changedRule.severity),
+          changedRule.params.stream().filter(p -> p.value != null).collect(Collectors.toMap(p -> p.key, p -> p.value)),
+          changedRule.templateKey,
+          changedRule.impacts == null ? new ArrayList<>() : changedRule.impacts.stream()
```

Comment:
instead of `new ArrayList<>()` you can use `Collections.emptyList()` or `List.of()`: it's immutable, and for instance `emptyList()` does not create new object on each call

Reply:
I will change those in the upcoming ticket since I would prefer not to run the pipelines again for it